docs/brillig/01_architecture.md

```diff
@@ -0,0 +1,26 @@
+# Brillig Architecture
+
+
+Finite Field VM
+---
+The Brillig VM operates on finite fields using a 128 bit integer representation, matching the native representation in Noir and its backends. This decision simplifies SNARK proving and optimizes for efficiency. The exact maximum integer value the field can store should not be relied upon, however, it is assured to be capable of packing 3 32-bit integers. When a BinaryIntOp is used, the field is first cast to a fixed bitsize integer that can be accommodated within the field, prior to performing operations. Certain operations, such as ordered comparison or signed division, only make sense in the context of BinaryIntOp.
```

Comment:
The finite field Brillig uses is generalizable based upon the field ACIR is using. Arkworks interfaces are used for the underlying Field interfaces. The maximum integer size supported is 128 bits, but Fields can be much large. ACIR currently lets you use 254 bit and 381 bit Fields (even though the only backend currently integrated into Noir supports the bn254 finite field). It would be more accurate to state something like: "Brillig VM operates over finite fields and supports using up to a 128 bit representation for integers. All integers ultimately translate to the finite field which Brillig is based upon. For example when a BinaryIntOp is used..."

Why does this decision simplify SNARK proving and optimize for efficiency? Due to all integers being based upon a finite field which is native to SNARKs. I think it is an unclear to say that in the beginning of the paragraph and perhaps would flow better if it came after mentioning that all integers translate to fields.